tests/test_admin.py

```diff
@@ -0,0 +1,167 @@
+            qs_default = response_default.context["cl"].queryset
+            response_unpublished = self.client.get(base_url + "?unpublished=1")
+            qs_unpublished = response_unpublished.context["cl"].queryset
+
+        self.assertEqual(set(qs_default), set(expected))
+        self.assertEqual(set(qs_unpublished), set(expected_unpublished))
+
+
+class DuplicateViewTestCase(CMSTestCase):
+    def test_obj_does_not_exist(self):
+        with self.login_user_context(self.get_superuser()), patch(
+            "djangocms_pageadmin.admin.PageContentAdmin._get_obj_does_not_exist_redirect",
+            return_value=HttpResponse(),
+        ) as mock:
+            response = self.client.get(
+                self.get_admin_url(PageContent, "duplicate_content", "foo")
+            )
+        mock.assert_called_once()
+        self.assertEqual(response, mock.return_value)
+
+    def test_get(self):
+        pagecontent = PageContentWithVersionFactory()
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.get(
+                self.get_admin_url(PageContent, "duplicate_content", pagecontent.pk)
+            )
+        self.assertEqual(response.status_code, 200)
+
+    def test_post_empty_slug(self):
+        pagecontent = PageContentWithVersionFactory()
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.post(
+                self.get_admin_url(PageContent, "duplicate_content", pagecontent.pk),
+                data={"slug": ""},
+            )
+            form = response.context["form"]
+        self.assertEqual(response.status_code, 200)
+        self.assertFalse(form.is_valid())
+        self.assertIn("slug", form.errors)
+        self.assertEqual(form.errors["slug"], ["This field is required."])
+
+    def test_post_empty_slug_after_slugify(self):
+        pagecontent = PageContentWithVersionFactory()
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.post(
+                self.get_admin_url(PageContent, "duplicate_content", pagecontent.pk),
+                data={"site": Site.objects.first().pk, "slug": "£"},
+            )
+            form = response.context["form"]
+        self.assertEqual(response.status_code, 200)
+        self.assertFalse(form.is_valid())
+        self.assertIn("slug", form.errors)
+        self.assertEqual(form.errors["slug"], ["Slug must not be empty."])
+
+    def test_post(self):
+        pagecontent = PageContentWithVersionFactory(template="page.html")
+        placeholder = PlaceholderFactory(slot="content", source=pagecontent)
+        add_plugin(placeholder, "TextPlugin", pagecontent.language, body="Test text")
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.post(
+                self.get_admin_url(PageContent, "duplicate_content", pagecontent.pk),
+                data={"site": Site.objects.first().pk, "slug": "foo"},
```

Comment:
I suggest putting some caps and spaces into the slug data here. That way you know slugify has definitely run when you do the asserts.